### ISSUE TITLE
Dangling symlinks cause crash

### ISSUE THREAD

**Issue by anacrolix** [anacrolix/dms#17]
Here's the relevant stacktrace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x429025]

goroutine 157 [running]:
runtime.panic(0x755280, 0xc2100d8190)
    /home/matt/src/go/src/pkg/runtime/panic.c:266 +0xb6
bitbucket.org/anacrolix/dms/dlna/dms.(*Server).fileEntries(0xc21008e780, 0x0, 0x0, 0xc2100f3f80, 0x36, ...)
    /home/matt/gopath/src/bitbucket.org/anacrolix/dms/dlna/dms/dms.go:313 +0x255
bitbucket.org/anacrolix/dms/dlna/dms.(*Server).childCount(0xc21008e780, 0xc2100f3f80, 0x36, 0xc2100b1680)
    /home/matt/gopath/src/bitbucket.org/anacrolix/dms/dlna/dms/dms.go:151 +0x1a2
bitbucket.org/anacrolix/dms/dlna/dms.(*Server).entryObject(0xc21008e780, 0x7fbc9d4b15b8, 0xc2100f2370, 0xc2100f3f80, 0x36, ...)
    /home/matt/gopath/src/bitbucket.org/anacrolix/dms/dlna/dms/dms.go:229 +0x19a
bitbucket.org/anacrolix/dms/dlna/dms.func·009(0x766d20, 0xc2100ffa80, 0xc210075038, 0x7fbc9d499f28)
    /home/matt/gopath/src/bitbucket.org/anacrolix/dms/dlna/dms/dms.go:379 +0x85
bitbucket.org/anacrolix/dms/futures.func·004(0xc210075010, 0x573ef0)
```

---
- Bitbucket: https://bitbucket.org/anacrolix/dms/issue/17
- Originally reported by: Matt Joiner
- Originally created at: 2013-09-17T16:01:27.057

**Comment by anacrolix** [comment]
_From David Mackay on 2013-09-17 19:47:51+00:00_

I also get this when I specify a path with no direct children that are media files.  The path that I'm specifying has sub folders with the media.

**Comment by anacrolix** [comment]
_From Matt Joiner on 2013-09-18 06:03:40+00:00_

I can't reproduce the problem you describe. Would you like to raise a ticket for this if it still exists after I fix this issue?

**Comment by anacrolix** [comment]
_From Matt Joiner on 2013-09-18 07:22:54+00:00_

Fix Issue #17: Stat errors place nil objects in the file info slice

→ <<cset 85cbf5ccf88b>>

**Comment by anacrolix** [comment]
_From David Mackay on 2013-09-18 14:22:31+00:00_

That fixed the issue, thanks.
